source/extensions/quic_listeners/quiche/platform/string_utils.cc

```diff
@@ -89,7 +89,9 @@ bool HexDecodeToUInt32(absl::string_view data, uint32_t* out) {
   std::string byte_string = absl::HexStringToBytes(data_padded);
 
   ASSERT(byte_string.size() == 4u);
-  *out = ntohl(*reinterpret_cast<const uint32_t*>(byte_string.c_str()));
+  uint32_t bytes;
+  memcpy(&bytes, byte_string.data(), byte_string.length());
```

Comment:
per discussions about memcpy issues can you change the above ASSERT to a RELEASE_ASSERT, or, use std::min(byte_string.length(), sizeof(bytes) as the 3rd arg for memcpy.

That way this will be obviously correct at a glance without worrying about what HexStringToBytes might return.

Reply:
or you can use MemBlockBuiilder in source/common/common/mem_block_builder.h

Reply:
done